### ISSUE TITLE
Confused regarding login duration and logging out

### ISSUE THREAD

**Issue by bleutzinn** [delight-im/PHP-Auth#164]
In [Keeping the user logged in](https://github.com/delight-im/PHP-Auth#keeping-the-user-logged-in) you write:
> Without the persistent login, which is the default behavior, a user will only stay logged in until they close their browser, or as long as configured via session.cookie_lifetime and session.gc_maxlifetime in PHP.

However it seems that persistent login is the default setting.

When I have  `session.cookie_lifetime = 0` and `session.gc_maxlifetime = 0` (both checked by examining `phpinfo()` ) and explicitly use NULL as the third parameter of `Auth#login` then closing the browser and restarting it half an hour later still keeps me logged in.

Also doing `$auth->login($userEmail, $password, 30);` does not help. The session is never terminated and the only way to log out is doing it manually.

I wonder how to limit a user's session duration to for example 20 minutes?

FYI in `app/index.php` I echo the value of `$app->auth()->isRemembered()` which always is `false` also after having closed the browser.

**Comment by bleutzinn** [comment]
Usually I use Safari for using different resources on the web. For testing etc. I use Chrome and Firefox.

The strange thing is that Safari is the only of these three in which the session is closed when the browser is restarted. It works as documented with and without a third parameter of `Auth#login`. When a duration is set with the third parameter and Safari is restarted before that period has elapsed the user is still logged in and `$app->auth()->isRemembered()` is `true`. This is also correct behaviour.

I'm totally lost why Chrome and Firefox both behave so differently and wrong. Although I'm not happy with this situation I think PHP-Foundation is working fine.

I sure hope someone can shed more light on this. Anyone?

**Comment by ocram** [comment]
First, while `session.cookie_lifetime = 0` is correct for what you want (i.e. sessions expiring when the browser is closed), I’m not sure if `session.gc_maxlifetime = 0` makes sense. Wouldn’t this cause PHP to treat sessions as garbage as soon as they are `0` seconds old (i.e. always)?

> I echo the value of `$app->auth()->isRemembered()` which always is `false`

Sounds good. That means you have indeed not been signed in using the long-lived “remember me” cookie but using the short-lived session cookie, unless there’s a bug in this library here.

The Safari behavior seems to be correct and underlines that this library may actually have a correct implementation.

You could inspect your cookies in the developer tools of your browser to check if the expected lifetime is set there.

When it comes to Firefox and Chrome, I guess it’s a browser configuration issue, as discussed [here](https://github.com/delight-im/PHP-Auth/issues/157).

Does that help?

**Comment by bleutzinn** [comment]
Yes your explanation and reference to issue [#157](https://github.com/delight-im/PHP-Auth/issues/157) did help.

Indeed setting `session.gc_maxlifetime = 0` does not make sense. I've read up on garbage collection and have some better understanding how things work.

The long-lived “remember me” cookie ("remember_<token>") has the correct time set for expire and the cookie is deleted when that time has been reached.

Maybe because I've made manually logging out kind of a habit I wasn't aware of the default settings in Firefox and Chrome that re-use a session after shutdown and startup. I changed that setting and sessions are now closed on browser close.

Again thanks for your help.
